tests/test_config.py

```diff
@@ -229,6 +229,45 @@ def test_local_creation_with_topdir():
     assert 'pytest' not in cfg(f=GLOBAL)
     assert cfg(f=LOCAL, topdir=str(topdir))['pytest']['key'] == 'val'
 
+def test_append():
+    update_testcfg('pytest', 'key', 'system', configfile=SYSTEM)
+    update_testcfg('pytest', 'key', 'global', configfile=GLOBAL)
+    update_testcfg('pytest', 'key', 'local', configfile=LOCAL)
+    # Appending with no configfile specified should modify the local one
+    cmd('config -a pytest.key ,bar')
+
+    # Only the local one will be modified
+    assert cfg(f=SYSTEM)['pytest']['key'] == 'system'
+    assert cfg(f=GLOBAL)['pytest']['key'] == 'global'
+    assert cfg(f=LOCAL)['pytest']['key'] == 'local,bar'
+
+    # Test a more complex one, and at a particular configfile level
+    update_testcfg('build', 'cmake-args', '-DCONF_FILE=foo.conf', configfile=GLOBAL)
+    assert cfg(f=GLOBAL)['build']['cmake-args'] == '-DCONF_FILE=foo.conf'
+
+    # Use a list instead of a string to get quoting right
+    cmd(['config', '--global', '-a', 'build.cmake-args', '--',
+         ' -DEXTRA_CFLAGS=\'-Wextra -g0\' -DFOO=BAR'])
```

Comment:
Nit, use outer `"` quotes so you don't need to escape the inner `'` ones?

```suggestion
         " -DEXTRA_CFLAGS='-Wextra -g0' -DFOO=BAR"])
```

Reply:
I had it like that before, but I personally actually prefer it this way, because it is consistent with the rest of the elements in the list.